src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -94,22 +97,22 @@ def setUpCouchDBReplication(self):
                                         'filter': "T0Request/repfilter"})
         else:
             # set up workqueue replication
-            wqfilter = 'WorkQueue/queueFilter'
-            parentQURL = self.config.WorkQueueManager.queueParams["ParentQueueCouchUrl"]
-            childURL = self.config.WorkQueueManager.queueParams["QueueURL"]
-            query_params = {'childUrl': childURL, 'parentUrl': sanitizeURL(parentQURL)['url']}
+            parentQueueUrl = self.workqueueSelector["WMCore\.WorkQueue\.DataStructs\.WorkQueueElement\.WorkQueueElement"]["ParentQueueUrl"]
             localQInboxURL = "%s_inbox" % self.config.AnalyticsDataCollector.localQueueURL
-            self.replicatorDocs.append({'source': sanitizeURL(parentQURL)['url'], 'target': localQInboxURL,
-                                        'filter': wqfilter, 'query_params': query_params})
-            self.replicatorDocs.append({'source': localQInboxURL, 'target': parentQURL,
-                                        'filter': wqfilter, 'query_params': query_params})
+            self.replicatorDocs.append({'source': parentQueueUrl,
```

Comment:
What if replicator docs already have such source or target? is it better to check it first. I know that CouchDB will create a revision, but may be you don't want that. It depends on logic of handling multiple versions of the same document.

Reply:
Replication documents are deleted before new replications are defined (see the beginning of this method).